enumerable.rb

```diff
@@ -0,0 +1,279 @@
+# frozen_string_literal: true
+# rubocop:disable all
+module Enumerable
+  def my_each
+    if block_given?
+      i = 0
+      while i < size
+        yield self[i]
+        i += 1
+      end
+    else
+      p 'No block given!'
+    end
+  end
+
+  def my_each_with_index
+    if block_given?
+      i = 0
+      while i < size
+        yield(self[i], i)
+        i += 1
+      end
+    else
+      p 'No block given!'
+    end
+  end
+
+  def my_select
+    if block_given?
+      array2 = []
+      i = 0
+      while i < size
+        array2 << self[i] if yield self[i]
+        i += 1
+      end
+    else
+      p 'No block given!'
+    end
+    array2
+  end
+
+  def my_all?
+    if block_given?
+      i = 0
+      while i < size
+        return false unless yield self[i]
+
+        i += 1
+      end
+    else
+      p 'No block given!'
+    end
+    true
+  end
+
+  def my_any?
+    if block_given?
+      i = 0
+      while i < size
+        return true if yield self[i]
+
+        i += 1
+      end
+    else
+      p 'No block given!'
+    end
+    false
+  end
+
+  def my_none?
+    if block_given?
+      i = 0
+      while i < size
+        return false if yield self[i]
+
+        i += 1
+      end
+    else
+      p 'No block given!'
+    end
+    true
+  end
```

Comment:
These methods should be able to take a parameter like the original methods,
To avoid this error. 

```
  ArgumentError:
        wrong number of arguments (given 1, expected 0)
```